sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-record.ts

```diff
@@ -0,0 +1,193 @@
+   * Also known as the Period of Study Commencement Date (PSCD).
+   */
+  educationalStartDate: Date;
+  /**
+   * Date that studies end. Generally derived from the latest application. Last day of classes or last
+   * exam date, whichever is latest. Also known as the Period of Study End Date (PSED).
+   */
+  educationalEndDate: Date;
+  /**
+   * Federal school code.
+   */
+  federalInstitutionCode: string;
+  /**
+   * Number of study weeks in study period.
+   */
+  weeksOfStudy: number;
+  /**
+   * Federal field of study code.
+   */
+  fieldOfStudy: number;
+  /**
+   * Student's Year of study.
+   */
+  yearOfStudy: number;
+  /**
+   * Date that the school confirmed enrolment.
+   */
+  enrollmentConfirmationDate: Date;
+  /**
+   * Student date of birth.
+   */
+  dateOfBirth: Date;
+  /**
+   * Student last name.
+   */
+  lastName: string;
+  /**
+   * Student first name.
+   */
+  firstName: string;
+  /**
+   * Student address.
+   */
+  addressLine1: string;
+  /**
+   * Student address
+   */
+  addressLine2: string;
+  /**
+   * Student city address.
+   */
+  city: string;
+  /**
+   * Student email address.
+   */
+  emailAddress: string;
+  /**
+   * Student gender M=Male F=Female.
+   */
+  gender: string;
+  /**
+   * Borrower marital status S= Single, M = Married, O = Other.
+   */
+  maritalStatus: string;
+  /**
+   * Student number (at the Educational Institution).
+   */
+  studentNumber: string;
+  /**
+   * Total of all grant awards on this Entitlement Record.
+   */
+  totalGrantAmount: number;
+  /**
+   * Certificate number, the generated sequence number for the file.
+   */
+  certNumber: number;
+  /**
+   * Total of all BC grant awards on this Entitlement Record.
+   */
+  totalBCSGAmount: number;
+  /**
+   * Total of all BC grant awards on this Entitlement Record.
+   */
+  totalCSGP_PTAmount: number;
+  /**
+   * Total of all BC grant awards on this Entitlement Record.
+   */
+  totalCSGP_PDAmount: number;
+  /**
+   * Total of all BC grant awards on this Entitlement Record.
+   */
+  totalCSGP_PTDEPAmount: number;
+
+  public getFixedFormat(): string {
+    const record = new StringBuilder();
+    record.append(this.recordType);
+    record.appendWithEndFiller(this.lastName, 25, SPACE_FILLER);
+    record.appendWithEndFiller(this.firstName || "", 15, SPACE_FILLER);
+    record.repeatAppend(SPACE_FILLER, 3); // Initials, optional, not provided.
+    record.append(this.sin, 9);
+    record.append(this.gender, 1);
+    record.appendDate(this.dateOfBirth, DATE_FORMAT);
+    record.appendWithEndFiller(this.maritalStatus, 4, SPACE_FILLER);
+    record.appendWithEndFiller(this.addressLine1, 40, SPACE_FILLER);
+    record.appendWithEndFiller(this.addressLine2 || "", 40, SPACE_FILLER);
+    record.repeatAppend(SPACE_FILLER, 40); // AddressLine 3, optional, not provided.
+    record.appendWithEndFiller(this.city, 25, SPACE_FILLER);
+    record.append("BC  ", 4); //TODO Province, is hardcoded to "BC  ".
```

Comment:
just for my understanding, why there are extra space in province and country

Reply:
@guru-aot I would suggest using the method to add the filler, even for contacts. It would be easier to read.
```typescript
record.appendWithEndFiller("BC", 4, SPACE_FILLER);
```

Reply:
Currently we have visibility for BC province and country Canada, as those codes are lesser than the length to be provided in the spacer, adding additional space in the suffix.